ngraph/frontend/onnx/frontend/src/edge_mapper.cpp

```diff
@@ -146,33 +150,38 @@ InputEdge onnx_editor::EdgeMapper::find_input_edge(const EditorNode& node, const
 }
 
 OutputEdge onnx_editor::EdgeMapper::find_output_edge(const EditorNode& node, const EditorOutput& out) const {
-    // identification can be both based on node name and output name
-    const auto& node_indexes = find_node_indexes(node.m_node_name, node.m_output_name);
-    int node_index = -1;
-    if (node_indexes.size() == 1) {
-        node_index = node_indexes[0];
-    } else if (node_indexes.empty()) {
-        throw ngraph_error("Node with name: " + (node.m_node_name.empty() ? "not_given" : node.m_node_name) +
-                           " and output_name: " + (node.m_output_name.empty() ? "not_given" : node.m_output_name) +
-                           " was not found");
-    } else if (!out.m_output_name.empty())  // output indexes are not deterministic if a node name is ambiguous
-    {
-        // many nodes with the same name
-        // check if some of found index matches output name
-        int matched_outputs_number = 0;
-        for (const auto& index : node_indexes) {
-            if (std::count(std::begin(m_node_outputs[index]), std::end(m_node_outputs[index]), out.m_output_name) > 0) {
-                node_index = index;
-                ++matched_outputs_number;
+    int node_index = node_index = node.m_node_index;
+    if (node_index == -1) {  // the node index is not provided
+        // identification can be both based on node name and output name (if the node index is not provided)
+        const auto& node_indexes = find_node_indexes(node.m_node_name, node.m_output_name);
+        if (node_indexes.size() == 1) {
+            node_index = node_indexes[0];
+        } else if (node_indexes.empty()) {
+            throw ngraph_error("Node with name: " + (node.m_node_name.empty() ? "not_given" : node.m_node_name) +
+                               " and output_name: " + (node.m_output_name.empty() ? "not_given" : node.m_output_name) +
+                               " was not found");
+        } else if (!out.m_output_name.empty())  // output indexes are not deterministic if a node name is ambiguous
+        {
+            // many nodes with the same name
```

Comment:
Is this condition covered by tests?

Reply:
yes, by:
- `onnx_editor.editor_api_select_output_edge_by_ambiguous_node_name_but_matched_output`
- `onnx_editor.editor_api_select_output_edge_by_ambiguous_node_name_and_not_matched_output`